api/client/client.go

```diff
@@ -266,3 +276,21 @@ func NewAuthServer(addr string, provider auth.Provider) *AuthRemote {
 func (ar *AuthRemote) Sign(req []byte) ([]byte, error) {
 	return ar.AuthSign(req, nil, ar.provider)
 }
+
+// nomalizeURL checks for http/https protocol, appends "http" as default protocol if not defiend in url
+func normalizeURL(addr string) string {
+	addr = strings.TrimSpace(addr)
+	u, err := url.Parse(addr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if !strings.Contains(u.Scheme, "http") {
```

Comment:
There's no need to convert to string and then reparse here, you can just check if `u.Scheme==""` and set it to `"https"`, then return the resulting `*url.URL` for later processing.